src/core/scalar/vector.rs

```diff
@@ -1096,3 +1098,532 @@ impl SignedVector<f32> for XYZF32A16 {
 
 impl SignedVector3<f32> for XYZF32A16 {}
 impl FloatVector3<f32> for XYZF32A16 {}
+
+impl<T> Not for XY<T>
```

Comment:
I have not implemented traits directly on the inner types because I can't do the same thing for SIMD types due to the orphan rule. So my preference would be to provide a trait that has all the required methods like the existing traits work in the core traits module.